lnrpc/neutrinorpc/neutrino_server.go

```diff
@@ -0,0 +1,428 @@
+		Confirmations:     resp.Confirmations,
+		StrippedSize:      resp.StrippedSize,
+		Size:              resp.Size,
+		Weight:            resp.Weight,
+		Height:            resp.Height,
+		Version:           resp.Version,
+		VersionHex:        resp.VersionHex,
+		Merkleroot:        resp.Merkleroot,
+		Time:              resp.Time,
+		Nonce:             resp.Nonce,
+		Bits:              resp.Bits,
+		Ntx:               resp.Ntx,
+		PreviousBlockHash: resp.PreviousBlockHash,
+		RawHex:            resp.RawHex,
+	}, nil
+}
+
+// GetBlock returns a block with a particular block hash. If the block is
+// found in the cache, it will be returned immediately. Otherwise a block will
+// be requested from the network, one peer at a time, until one answers.
+//
+// NOTE: Part of the NeutrinoKitServer interface.
+func (s *Server) GetBlock(ctx context.Context,
+	in *GetBlockRequest) (*GetBlockResponse, error) {
+
+	if s.cfg.NeutrinoCS == nil {
+		return &GetBlockResponse{}, ErrNeutrinoNotActive
+	}
+
+	var hash chainhash.Hash
+	if err := chainhash.Decode(&hash, in.Hash); err != nil {
+		return &GetBlockResponse{}, err
+	}
+
+	return s.getBlock(hash)
+}
+
+// GetCFilter returns a compact filter of a particular block.
+// If found, only regular filters will be returned.
+//
+// NOTE: Part of the NeutrinoKitServer interface.
+func (s *Server) GetCFilter(ctx context.Context,
+	in *GetCFilterRequest) (*GetCFilterResponse, error) {
+
+	if s.cfg.NeutrinoCS == nil {
+		return &GetCFilterResponse{}, ErrNeutrinoNotActive
+	}
+
+	var hash chainhash.Hash
+	if err := chainhash.Decode(&hash, in.Hash); err != nil {
+		return &GetCFilterResponse{}, err
+	}
+
+	// GetCFilter returns a compact filter from the database. If it is missing,
+	// it requests the compact filter from the network.
+	filer, err := s.cfg.NeutrinoCS.GetCFilter(hash, wire.GCSFilterRegular)
+	if err != nil {
+		return &GetCFilterResponse{}, err
+	}
+
+	filterlBytes, err := filer.Bytes()
+	if err != nil {
+		return &GetCFilterResponse{}, err
+	}
+
+	return &GetCFilterResponse{Filter: filterlBytes}, nil
+}
+
+func (s *Server) getBlock(hash chainhash.Hash) (*GetBlockResponse, error) {
+	block, err := s.cfg.NeutrinoCS.GetBlock(hash)
+	if err != nil {
+		return &GetBlockResponse{}, err
+	}
+
+	header, _, err := s.cfg.NeutrinoCS.BlockHeaders.FetchHeader(&hash)
+	if err != nil {
+		return &GetBlockResponse{}, err
+	}
+
+	blockData, err := block.Bytes()
+	if err != nil {
+		return &GetBlockResponse{}, err
+	}
+
+	strippedData, err := block.BytesNoWitness()
+	if err != nil {
+		return &GetBlockResponse{}, err
+	}
+
+	bestBlock, err := s.cfg.NeutrinoCS.BestBlock()
+	if err != nil {
+		return &GetBlockResponse{}, err
+	}
+
+	// Transaction weight is calculated as
+	// base transaction size * 3 + total transaction size.
+	weight := int64(len(strippedData)*3 + len(blockData))
+
+	// Convert txids to a string array.
+	transactions := block.Transactions()
+	var tx = make([]string, len(transactions))
```

Comment:
var statement can be dropped here.

Reply:
Sure thing.

Reply:
You're right. I missunderstood and removed the comment. Will remove it tomorrow.

Reply:
Merged.